content/shmem_info_get_name.tex

```diff
@@ -19,17 +19,18 @@
 
 \apidescription{
     This routine returns the vendor defined character string of size defined by
-    the constant SHMEM\_MAX\_NAME\_LEN. The program calling this function
-    prepares the memory of size SHMEM\_MAX\_NAME\_LEN, and the implementation
-    copies the string of size at most SHMEM\_MAX\_NAME\_LEN. In C, the string is
-    terminated by a null character.  In Fortran, the string of size less than
-    SHMEM\_MAX\_NAME\_LEN is padded with blank characters up to size
-    SHMEM\_MAX\_NAME\_LEN. The implementation copying a string of size greater
-    than SHMEM\_MAX\_NAME\_LEN results in an undefined behavior.  Multiple
-    invocations of the routine in an \openshmem{} program always return the
-    same string.  For a given library implementation, the major and minor
-    version returned by these calls is consistent with the compile-time
-    constants defined in its shmem.h.
+    the library constant \CONST{SHMEM\_MAX\_NAME\_LEN}. The program calling this function
+    prepares the \VAR{name} memory buffer of at least size \CONST{SHMEM\_MAX\_NAME\_LEN}.
+    The implementation
+    copies the vendor defined string of size at most \CONST{SHMEM\_MAX\_NAME\_LEN} to \VAR{name}.
+    In \CorCpp{}, the string is
+    terminated by a null character.  In \Fortran{}, the string of size less than
+    \CONST{SHMEM\_MAX\_NAME\_LEN} is padded with blank characters up to size
+    \CONST{SHMEM\_MAX\_NAME\_LEN}.
+    If the \VAR{name} memory buffer is provided with size less than
+    \CONST{SHMEM\_MAX\_NAME\_LEN}, behavior is undefined.
+    For a given library implementation, the vendor string
+    returned is consistent with the library constant \CONST{SHMEM\_VENDOR\_STRING}.
```

Comment:
IIRC, the rationale for the existing get_name API was that the library may create that string during the call (e.g. based on dynamic linking with other parts of the communication stack).  I agree that what you suggested is much easier to use given the clarified definition we're currently working with (that it should be the same as the constant).